server/eclipse-project/src/main/java/us/freeandfair/corla/json/ContestToAuditJsonAdapter.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Free & Fair Colorado RLA System
+ * 
+ * @title ColoradoRLA
+ * @created Jul 28, 2017
+ * @copyright 2017 Free & Fair
+ * @license GNU General Public License 3.0
+ * @author Daniel M. Zimmerman <dmz@freeandfair.us>
+ * @description A system to assist in conducting statewide risk-limiting audits.
+ */
+
+package us.freeandfair.corla.json;
+
+import java.io.IOException;
+
+import com.google.gson.JsonSyntaxException;
+import com.google.gson.TypeAdapter;
+import com.google.gson.stream.JsonReader;
+import com.google.gson.stream.JsonWriter;
+
+import us.freeandfair.corla.model.Contest;
+import us.freeandfair.corla.model.ContestToAudit;
+import us.freeandfair.corla.model.ContestToAudit.AuditReason;
+import us.freeandfair.corla.model.ContestToAudit.AuditType;
+import us.freeandfair.corla.persistence.Persistence;
+
+/**
+ * JSON adapter for contest to audit information.
+ * 
+ * @author Daniel M. Zimmerman
+ * @version 0.0.1
+ */
+// the default constructor suffices for type adapters
+@SuppressWarnings("PMD.AtLeastOneConstructor")
+/**
+ * JSON adapter for the internal map from contests to choices.
+ */
+public final class ContestToAuditJsonAdapter 
+    extends TypeAdapter<ContestToAudit> {
+  /**
+   * The "contest" string (for JSON serialization).
+   */
+  private static final String CONTEST = "contest";
+  
+  /**
+   * The "reason" string (for JSON serialization).
+   */
+  private static final String REASON = "reason";
+  
+  /**
+   * The "audit type" string (for JSON serialization).
+   */
+  private static final String AUDIT = "audit";
+  
+  /**
+   * Writes a contest to audit object.
+   * 
+   * @param the_writer The JSON writer.
+   * @param the_info The object to write.
+   */ 
+  @Override
+  public void write(final JsonWriter the_writer, 
+                    final ContestToAudit the_contest) 
+      throws IOException {
+    the_writer.beginObject();
+    the_writer.name(CONTEST).value(the_contest.contest().id());
+    the_writer.name(AUDIT).value(the_contest.audit().toString());
+    if (the_contest.reason() != null) {
+      the_writer.name(REASON).value(the_contest.reason().toString());
+    }
+    the_writer.endObject();
+  }
+  
+  /**
+   * Checks the sanity of a contest to audit.
+   * 
+   * @param the_id The contest ID.
+   * @param the_reason The reason for audit.
+   * @param the_audit_type The audit type.
+   * @return the corresponding contest, if the data is sane, or null if
+   * the data is not.
+   */
+  private Contest sanityCheck(final Long the_id, 
```

Comment:
> then the model classes are explicitly (albeit statically) talking to the persistence infrastructure; that's kind of the way it was before when I had that scattered across all those model classes.

Ah, I see. Because the models subclass from `PersistentEntity`, and have some persistence-related annotations, I didn't realize you were working to decouple them.

I'll let you mull over the big picture, but if you're trying to stick to a decoupled "data mapper" design, that seems good to me. I guess those validity checks that are dependent on persistence/the state of the world are an annoying edge case.